k8s/14.md

```diff
@@ -0,0 +1,240 @@
+configmap/prometheus-kube-prometheus-prometheus                          1      40s
+configmap/prometheus-kube-prometheus-proxy                               1      40s
+configmap/prometheus-kube-prometheus-scheduler                           1      40s
+configmap/prometheus-kube-prometheus-workload-total                      1      40s
+configmap/prometheus-prometheus-kube-prometheus-prometheus-rulefiles-0   29     36s
+configmap/redis-config                                                   1      18s
+```
+
+</details>
+
+## Grafana dashboards
+
+In my case the command to run the service is `minikube service prometheus-grafana` (since `RELEASE_NAME` for me is `prometheus`).
+
+The default login credentials can be found from `<RELEASE_NAME>_grafana` secret, which has them base64-encoded.
+
+Here is a shortcut to decode values with the use of `jq`:
+
+```sh
+kubectl get secret prometheus-grafana -o jsonpath="{.data}" | \
+jq 'map_values(.|@base64d)'
+```
+
+```text
+{
+  "admin-password": "prom-operator",
+  "admin-user": "admin",
+  "ldap-toml": ""
+}
+```
+
+### CPU and memory usage of the StatefulSet of the app
+
+No data for usage for some reason. At least the quotas are shown.
+
+![CPU](https://user-images.githubusercontent.com/29694249/205436844-ed9a7e10-9787-4162-b04a-e3953cff4da2.png)
+![Memory](https://user-images.githubusercontent.com/29694249/205436852-05008fb8-cbdc-4838-9e9b-4ad6ccd781fd.png)
+
+### Most CPU-using Pod in the default namespace
+
+Well, also no usage here. But the quota on app pods is higher than on the Prometheus and Alertmanager.
+
+![image](https://user-images.githubusercontent.com/29694249/205436948-591346a8-6e72-47fc-a862-4370aa5466c9.png)
+
+### Node memory consumption and CPU usage
+
+No memory info for the node and only the quotas in k8s `Compute Resources`.
+
+![node](https://user-images.githubusercontent.com/29694249/205437221-e2bcdb52-24db-4e28-a48e-5c3b2f421d59.png)
+
+However, `Node Exporter` dashboard shows all the necessary hardware stats:
+
+![node-exporter](https://user-images.githubusercontent.com/29694249/205438164-cb2a5eb0-5aef-4d8a-ad6d-a70db27aebc9.png)
+
+### Kubelet Pods and containers count
+
+![image](https://user-images.githubusercontent.com/29694249/205437377-adcb5b33-cacb-4974-943b-b7ab51e517cb.png)
+
+### Most network-using Pod in the default namespace
+
+The data is missing in both k8s `Compute Resources` and `Networking` dashboards. I don't get why.
+
+![image](https://user-images.githubusercontent.com/29694249/205437451-616a22c7-aaf4-48b9-8c79-966740e6bfe5.png)
+
+![image](https://user-images.githubusercontent.com/29694249/205437940-5d788cc4-32d6-468f-acda-c14d8fd0fd0b.png)
+
+### Alert count
+
+![image](https://user-images.githubusercontent.com/29694249/205438318-5e8119db-40ad-453d-88fb-19ea6a4c64a5.png)
+
+## Init containers
+
+### Add volumes
+
+First, we need to add a new volume:
+
+```diff
+  volumes:
++   - name: init-dir
```

Comment:
wow that's a nice feature ![](https://steamcommunity-a.akamaihd.net/economy/emoticon/:ohh_yeah:)